test/mjsonwp/mjsonwp-specs.js

```diff
@@ -217,18 +217,32 @@ describe('MJSONWP', async () => {
         json: {id: 'baz', sessionId: 'lol', value: ['a']}
       });
 
+      // make sure adding the optional 'id' doesn't clobber a route where we
+      // have an actual required 'id'
       await request({
-        url: 'http://localhost:8181/wd/hub/session/foo/element/bar/value',
+        url: 'http://localhost:8181/wd/hub/session/foo/frame',
         method: 'POST',
         json: {id: 'baz'}
+      });
+    });
+
+    it('should properly handle variable payload for set value call', async () => {
+      await request({
+        url: 'http://localhost:8181/wd/hub/session/foo/element/bar/value',
+        method: 'POST',
+        json: {sessionId: 'lol'}
       }).should.eventually.be.rejectedWith("400");
 
-      // make sure adding the optional 'id' doesn't clobber a route where we
-      // have an actual required 'id'
       await request({
-        url: 'http://localhost:8181/wd/hub/session/foo/frame',
+        url: 'http://localhost:8181/wd/hub/session/foo/element/bar/value',
         method: 'POST',
-        json: {id: 'baz'}
+        json: {sessionId: 'lol', value: ['a'], text: 'bla'}
+      });
+
+      await request({
+        url: 'http://localhost:8181/wd/hub/session/foo/element/bar/value',
+        method: 'POST',
+        json: {sessionId: 'lol', text: 'bla'}
```

Comment:
likewise let's assert that the response is `bla`. (we use a FakeDriver for these tests, which just returns the text value so we can verify it got in correctly)